homeassistant/components/fastdotcom/sensor.py

```diff
@@ -38,9 +38,10 @@ class SpeedtestSensor(RestoreEntity, SensorEntity):
     _attr_icon = "mdi:speedometer"
     _attr_should_poll = False
 
-    def __init__(self, speedtest_data: dict[str, Any]) -> None:
+    def __init__(self, entry_id: str, speedtest_data: dict[str, Any]) -> None:
         """Initialize the sensor."""
         self._speedtest_data = speedtest_data
+        self._attr_unique_id = entry_id
 
     async def async_added_to_hass(self) -> None:
```

Comment:
Probably it would be better in a follow-up PR to check if we have data in `__init__.py` and raise `ConfigEntryError` if not so we can then remove `RestoreEntity` here completely and just get the data we already have.

Reply:
It is already running on setup hence why I write like I do as data does already exist.
The alternative what you write e.g. don't have an initial data collection

Reply:
Probably it would be better to more or less copy the `speedtestsensor` using a `DataUpdateCoordinator`

Reply:
I had a DataUpdateCoordinator in place in a a previous one. I suppose we have a good state now to fully implement it?